inmem_store.go

```diff
@@ -82,6 +82,10 @@ func (i *InmemStore) DeleteRange(min, max uint64) error {
 		delete(i.logs, j)
 	}
 	i.lowIndex = max + 1
```

Comment:
Let's assume that DeleteRange will delete a prefix of the log, a suffix of the log, neither, or the entire thing. I'm not a unit test, but I don't think this updates lowIndex and highIndex correctly when deleting a suffix, either before or after your change.